pandas/core/indexes/numeric.py

```diff
@@ -31,6 +31,7 @@ class NumericIndex(Index):
 
     """
     _is_numeric_dtype = True
+    _is_dotable = False  # Can't contain Python identifiers
```

Comment:
note: DatetimeIndex and friends inherit from Int64Index (which inherits from here).